### DISCUSSION TITLE
[Feature] Ability to specify SYNC destination album

### DISCUSSION THREAD

**Discussion by rkdships** [immich-app/immich#12748] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Immich has recently implemented a very long awaited feature: Sync Albums.

This feature however creates new albums on the backend. I already have about 500+ albums, some named exactly as folders on my phone.

It would be great if we were able to select desination albums so that assets are automatically uploaded into existing albums or albums that we select.

<hr>

How the Sync Albums feature fails right now: I have a 'Camera' album with 100,000+ assets. When I use a sync feature it creates a new 'Camera' album and now I have two: Camera (100,000+) and Camera (52). 52 is just a random number of assets uploaded since the start of using Sync Albums feature. Selecting a destination album would solve this issue.

Another use of destination albums. I have about 500 albums, most of them named in a pattern like 'country/cityName' or 'pictures/socialMediaApps/appName'. I would like to keep this pattern, but would love to see Sync Album feature just allow me to provide an album name (or ID?) to upload assets to for every source folder.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Arti1982** [comment]
I agree, I have several cell phones, and the folder name is always 'Camera'. So I am unable to sort them by album on Immich..

**Comment by kawazoe** [comment]
I would also like this feature. For me, I often take pictures from the picker in other apps, like Telegram or Discord to share with my friends. On my phone, these pickers save pictures into their own folders which can't be renamed. As a result, I have a "Camera" folder, a "Telegram Images" folder, and a "Pictures" folder. While I'll never be able to rename them, it would be useful if I could at least rename the albums, or even better merge them into a single album.

**Comment by LaVINCENTgit** [comment]
hello, It would be great to be able to choose in which of existing album on the server upload video and photos. For example, after a travel i would use the existing immich feature "share" directly from the phone gallery to upload pics on the server, but wuould be great to have the possibility to choose directly in which albut upload memories. Thanks a lot!

Edit: this possibility already exists for the web version... i'm referring, of course, to the mobile app

**Comment by crash1020** [comment]
Additionally, I would appreciate it if shared albums could be defined as a destination, or alternatively, shared albums with the same name could be displayed in one album. 
My wife and I both take pictures on vacation, for example. We both organize our folders in the iOS Photos app into separate albums. When uploading, these end up in two separate folders in our respective accounts. 
If both folders are shared with each other, two separate albums with the same name remain.
If these two were merged, that would be great. Alternatively, one could also define a shared folder as the upload destination.

**Comment by Orange-Icepop** [comment]
A similar problem with iOS. The "Recent" album is really a non-distinguishable album name, which makes me headache. It would be really helpful to add this feature.

**Comment by kaelnn** [comment]
I'm fairly new to Immich, but I'm already impressed -- this is a fantastic project that truly delivers on self-hosted photo and video backup with excellent performance, privacy, and usability.

I'd like to voice my strong support for this feature request: the ability to specify a **SYNC destination album**. This would greatly reduce redundant albums like *Camera (52)* appearing alongside the original *Camera* when syncing system-generated mobile albums.

**My perspective:**
- On mobile platforms, system-generated albums like *Camera*, *Screenshots*, or *Live Photos* can't be renamed by users. This often results in fragmented and messy album organization, especially across multiple devices.  
- I've noticed that a similar capability exists in the CLI through the `immich upload -A/--album-name` option, but this isn't something that can be done easily on mobile or tablet devices. Having the same flexibility on the mobile app would make album organization much cleaner and have a big impact for users who mainly rely on mobile uploads.

**Additional idea:**  
Immich already provides a powerful *storage template engine* for backend storage paths.  
It might be interesting to extend a similar concept to **album naming on sync**. For example, allowing per-device templates to define a target album format, such as:  

- Template: `{{variables}}-{{album}}`  
- Per-device examples:  
  - `{{deviceName}} {{album}}` → *Alice's iPhone Camera*, *Alice's iPad Screenshots*  
  - `Work {{album}}` → *Work Camera*, *Work Live Photos*
  - `{{y}}-{{device}}-{{album}}` → *2025-iPhone-Camera*, *2025-iPhone-Live Photos*

Thanks to the team for building such a compelling tool. I'm excited to see how this feature evolves!

**Comment by grillop** [comment]
I have installed and am very impressed with the product.  Given my current use case - it appears that I am possibly facing a deal breaker (for me of course).

I and my wife take pictures and we view them as a whole - that is completely shared.  So, I have 2 options.
1) Create separate accounts and have auto backup and then share those accounts.
    Problem here is that facial recognition does not take place across accounts (understandable..) but really are requirement for family
2) use the same account (no problem here) except we truly want to understand who took/owned the picture.  If we both upload our camera album,they are merged and that information is lost.

Thus, this feature request is very important to us, as we would simply choose a different destination album on the server and all is well.

Not clear we are ready to use Immich as a replacement - but until we can achieve a solution that does not require constant monitoring and manual uploads - we probably would be using Immich only as a backup repository (which we are grateful for).

Please consider adding this feature ...

I too, thank the team for all the hard work, and felt a bit hesitant to post this - as I do appreciate the tremendous work done here.

**Comment by Balthxzar** [comment]
A touch over a year later and no comment from anyone about this....

Given most phones don't give users the option to change the local directory, and some people (shock horror!) have multiple phones, it's incredibly cumbersome to have several different albums on Immich that are essentially the same thing. 

Having a Samsung and Sony phone myself, I have 3 (yes, three) different local folders that all make up my "camera roll" (PHOTOGRAPHY_PRO, VIDEOGRAPHY_PRO and Camera) - currently these all just exist as their own completely separate albums on Immich. 

Combine this with the same level of indifference on having "nested" albums, there's absolutely no way to combine these into a single album on Immich. 

The new fancy UI is very cool looking! Unfortunately it doesn't solve some pretty basic functional issues. 

What would it take to get ANY kind of response on this beyond the absolutely *hilarious* "any updates" post from a certain maintainer?

**Reply by bo0tzz** [reply]
What sort of response are you hoping to extract? We've got tons of issues and feature requests to work on, as well as things that aren't expressed in either of those. Plenty of feature requests will sit open without "response" until such a time as someone decides to work on it. Complaining about that won't speed anything up.

**Reply by Balthxzar** [reply]
"Hey this is something we think would work well, but we don't have the time to implement it now"

"Yeah so this wouldn't work for X, Y, Z reasons" 

"We don't plan to implement this as it doesn't follow our vision for the app" 

"Yeah, we're planning on adding this you should see it in a couple years"


Well, there's 4 to choose from.....

Ignoring these threads won't make them go away, and even if you don't have time to address them yet, don't be surprised when people ask why you haven't addressed them.......

**Reply by zackpollard** [reply]
We aren't trying to make them go away, we also don't have time to address every feature request that is made. You're one of the only people to have commented something like this on their request, so I think we will continue focusing on building things rather than replying to every request that is made.

**Comment by rkdships** [comment]
Fix that might work for this is matching by album name instead of creating a new one with same name but different ID. Can be a checkbox option on upload screen in the mobile app.

**Reply by grillop** [reply]
Doesn't help my situation in which my partners album had the same name as my album. Prefer to provide destination album names.  For me, I just want to differentiate.  Obviously, easier said than done and not a very high priority.

**Comment by floppymonster** [comment]
+1 this would be a great feature. we share an account (that we do this already indicates user management/sharing isnt fully op to snuff......) and thanks to having android and iphone we can sync to different albums, but it would be great if they could be namend "my android" , "your iphone" just  so you know who is the "owner". (and no seperate accounts with seperate librarys is something we do not like so will not use)

**Comment by AMP3D** [comment]
Has anyone found a reasonable work-around to this? I'm not sure if we all have unique use-cases here or what, but it seems Immich has a blind-spot when it comes to partner sharing while trying to keep a minimalist footprint. I've spent hours getting things configured to a somewhat reasonable state only to hit this problem. Would be nice if I can cross this last thing off my list. We're only backing up from a single device right now because this issue is a blocker. 

The only solution I can think of is having a third party app like Syncthing do the backups instead to my external library folders, but that seems so lame when this is somewhat low hanging fruit. I'd be happy to hear other ideas.

**Reply by floppymonster** [reply]
Our workaround is to just upload with the device album name, since 2 phones are all we really need and I am on android and my wife is on iphone with a proper setup of the apps. The "recent" album are all photos from her and the "camera" album are all from me. Using a single account. 

I have decided to not use any external library and moved all old photos over in these folder (mine to "camera" and my wifes to "recent") with the immich mass upload CLI tool. 

When setting it up I did not like it that much. But been using it for months now and not even thinking about it anymore.

**Reply by AMP3D** [reply]
Aye if that were the case for me, it would be a non-issue. My wife and I use the exact same phone, so that unfortunately doesn't work. I tried syncthing-fork as a workaround, but the battery drain is too much, even when conservatively configured.

**Reply by oliverrahner** [reply]
I have the same issue as some others here: 
I'm sharing the same account with my wife (mainly because faces/people don't work using partner sharing, which I'd love otherwise), but I still want to know which photo has been uploaded from which device.

I came up with the following concept (aided by Claude):
I use the _hostname_ I set up in each iOS app to separate and then have a small proxy shim (Go based) that takes these hostnames to manipulate the target album name.
So, my immich instance runs on `photos.example.com`.
I point my iOS app at `oliver.photos.example.com` and my wife's app at `hername.photos.example.com`. The shim then takes these incoming host names from the HTTP `Host` header and changes the target album to `Oliver` and `HerName`.

I will test drive this for some time and then see if it's stable enough or has any bad side effects, but right now I think it's a reasonable effort, at least if you're into tinkering :) I'll probably also publish the shim later.

**Reply by oliverrahner** [reply]
Here is the proxy I'm using:
https://github.com/oliverrahner/immich-upload-shim

**Reply by Balthxzar** [reply]
Brilliant work, it seems like a stupidly simple solution (though not one I could have implemented) 

Hopefully the Immich team can take some notes and finally add this functionality. In the time I was waiting, I ended up getting another Samsung phone which actually solved the issue, perhaps the maintainers can add "just buy matching devices" to the accepted solution"  🥴

**Reply by grillop** [reply]
I second that. I plan to give this a go. I use zoraxy reverse proxy and pretty sure it will work out.  Must be some fundamental structural reasons that a change to allow configuring a server album to accept uploads, but apparently it's not so easy. 

I'll report back when I get to it. 

Many thanks

**Comment by borgqueenx** [comment]
also would like this feature...
